src/main/java/uk/gov/hmcts/ccd/domain/service/accessprofile/filter/RoleAssignmentsFilteringServiceImpl.java

```diff
@@ -0,0 +1,71 @@
+package uk.gov.hmcts.ccd.domain.service.accessprofile.filter;
+
+import java.util.List;
+import java.util.stream.Collectors;
+import lombok.extern.slf4j.Slf4j;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.stereotype.Component;
+import uk.gov.hmcts.ccd.domain.model.casedataaccesscontrol.RoleAssignmentFilteringResult;
+import uk.gov.hmcts.ccd.domain.model.casedataaccesscontrol.RoleAssignments;
+import uk.gov.hmcts.ccd.domain.model.casedataaccesscontrol.RoleMatchingResult;
+import uk.gov.hmcts.ccd.domain.model.definition.CaseDetails;
+import uk.gov.hmcts.ccd.domain.model.definition.CaseTypeDefinition;
+import uk.gov.hmcts.ccd.domain.model.std.CaseDataContent;
+import uk.gov.hmcts.ccd.domain.service.AccessControl;
+import uk.gov.hmcts.ccd.domain.service.accessprofile.filter.matcher.AttributeMatcher;
+
+@Slf4j
+@Component
+public class RoleAssignmentsFilteringServiceImpl implements RoleAssignmentsFilteringService, AccessControl {
+
+    private List<AttributeMatcher> attributeMatchers;
+
+    @Autowired
+    public RoleAssignmentsFilteringServiceImpl(List<AttributeMatcher> attributeMatchers) {
+        this.attributeMatchers = attributeMatchers;
+    }
+
+    @Override
+    public List<RoleAssignmentFilteringResult> filter(RoleAssignments roleAssignments,
+                                                      CaseDetails caseDetails) {
+        log.info("Filter role assignments for case {}", caseDetails.getReference());
+
+        List<RoleAssignmentFilteringResult> roleAssignmentFilterResults = roleAssignments
+            .getRoleAssignments()
+            .stream()
+            .map(roleAssignment -> {
+                RoleAssignmentFilteringResult result = new RoleAssignmentFilteringResult(roleAssignment,
+                    new RoleMatchingResult());
+
+                attributeMatchers.forEach(matcher -> matcher.matchAttribute(result, caseDetails));
```

Comment:
minor, we can leave it. 
It's not a good practice using input-output parameters like `result`. We could have passed in the role and the case details, and the matcher could have returned a match result